app/server.py

```diff
@@ -84,25 +84,25 @@ def update_product(id):
   """
   data = inventory.get_product(id)
   info = request.get_json()
-  total = int(data[USED]) + int(data[NEW]) + int(data[OPEN_BOX])
-  prod_type = info[TYPE]
 
   if data is None:
     return make_response("Product not found", HTTP_404_NOT_FOUND)
 
-  elif not utils.is_valid(info):
+  if not utils.is_valid(info):
     return make_response("Product data is not valid", HTTP_400_BAD_REQUEST)
 
-  elif total - data[prod_type] + info[QUANTITY] > int(data[RESTOCK_LEVEL]):
+  total = int(data[USED]) + int(data[NEW]) + int(data[OPEN_BOX])
+  prod_type = info[TYPE]
+
+  if total - int(data[prod_type]) + int(info[QUANTITY]) > int(data[RESTOCK_LEVEL]):
     return make_response("Product amount exceed restock level", HTTP_400_BAD_REQUEST)
 
-  elif info[QUANTITY] < 0:
+  if int(info[QUANTITY]) < 0:
     return make_response("Product amount below zero", HTTP_400_BAD_REQUEST)
 
-  else:
-    data[prod_type] = info[QUANTITY]
-    inventory.put_product(id, data)
-    return make_response(jsonify(data), HTTP_200_OK)
+  data[prod_type] = info[QUANTITY]
```

Comment:
I am not sure.. should I change it? It's gonna be called with `self.redis.hmset(id, data)` and `jsonify(data)`, will the int type work ok with them?